tendermint-rs/src/amino_types/block_id.rs

```diff
@@ -18,6 +19,19 @@ impl block::ParseId for BlockId {
     }
 }
 
+impl ConsensusMessage for BlockId {
+    fn validate_basic(&self) -> Result<(), ValidationError> {
+        // Hash can be empty in case of POLBlockID in Proposal.
+        if !self.hash.is_empty() && self.hash.len() != SHA256_HASH_SIZE {
+            return Err(ValidationError::new(ValidationErrorKind::InvalidHashSize));
```

Comment:
Thanks! Makes sense, I'll implement the From trait and go with this more succinct here and everywhere else.

Reply:
One way you can further reduce the boilerplate is to import the variants of `ValidationErrorKind`, e.g.:

```
use super::validate::ValidationErrorKind::{InvalidHashSize, NegativePOLRound};
```

or

```
use super::validate::ValidationErrorKind::*;
```

Then you can refer to the variants directly by their names, rather than always having to include the `ValidationError::*` prefix.